### PR TITLE
Updating doc for CMET

### PR DESCRIPTION
**Pull request by Aboli-msft**

### CODE REVIEW

File: 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md
Comment (VamseeInala) [inline 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md:1]: Medatadata -> Metadata

Comment (VamseeInala) [reply 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md:1]: 2 places

**Review by VamseeInala — APPROVED** [review]

File: 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md:1]: Corrected

File: 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+5. Toggle edit button to enable editing.
+
+![Resources](../../Images/06_BulkEdit_ToggleEdit.png)
+
+6. Enter the value for custom tag and click on 'Update'.
+  >Note: Tags must be PascalCase, starting with a letter and allowing alphanumeric characters, underscores, and hyphens. Length should be 3-50 characters.
+
+![Resources](../../Images/06_BulkEdit_Update.png)
+
+7. Close the edit popup after bulk updates success message is displayed.
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md:51]: After the success message for bulk updates is displayed, please close the edit popup.

File: 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+![Resources](../../Images/06_BulkEdit_UpdateSuccess.png)
+
+8. Currently after adding the tags to required controls, Baseline control inventory needs to be triggered from Metadata Aggregator function list.
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md:55]: Currently, after adding the tags to the necessary controls, you need to trigger the Baseline control inventory from the Metadata Aggregator function list.

File: 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md [hunk 3]
```diff
@@ -53,10 +60,10 @@ Above mentioned appsettings can be configured either from Azure portal or using
   - Invoke the configuration cmdlet
     - Configure-CustomControlAdditionPrerequisites -SubscriptionId "AzTSHostingSubscriptionId" -ScanHostRGName "AzTSHostingRGName"
 
------
+
 ## Access to CMET (Control metadata editor tool)
 
-Any user can add new controls using the CMET (Control metadata editor tool) from the AzTS UI. Only the privileged users can access CMET. To elevate any user to privileged role kindly follow the steps mentioned below:
+Only the privileged users can access CMET to update control metadata or to add new control. To elevate any user to privileged role kindly follow the steps mentioned below:
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md:66]: Only privileged users have access to CMET for updating control metadata or adding new controls. To elevate a user to a privileged role, please follow the steps outlined below:

File: 06-Customizing AzTS for your org/Extending AzTS/Readme.md [hunk 4]
```diff
@@ -2,19 +2,24 @@
...
-Azure Tenant Security Solution (AzTS) provides capability to add new controls(for existing services supported by AzTS) to customize the AzTS for your organization as per your need.
-Currently you can extend AzTS controls set by either adding a new control based on custom Azure Policy or based on Microsoft Defender for Cloud assessment using Control Metadata Editor Tool. 
 
-## Enabling Control Medatadata Editor tool
-This feature is not enabled by default. To enable this feature for your AzTS setup, please follow steps mentioned [here](Prerequisites.md#prerequisite-azts-configurations-to-enable-control-medatadata-editor-toolcmet).
+## Customizing AzTS for your org
 
-## Add new control for existing services
+### Control Metadata Editor tool
+CMET (Control Metadata Editor Tool) is a tool that can be used by privileged users (mostly by admins maintaining AzTS setup for organization)to customize AzTS for organization. It provides capabilities to update control metadata (Tags, Display Name, control settings(if applicable)), enable or disable control, etc.
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/Extending AzTS/Readme.md:9]: CMET (Control Metadata Editor Tool) is a tool designed for privileged users, typically administrators responsible for maintaining the AzTS setup for an organization. It allows customization of AzTS by providing features such as updating control metadata (tags, display name, control settings if applicable), enabling or disabling controls, and more.

Please note that this feature is disabled by default. To enable CMET for your AzTS setup, kindly follow the steps mentioned here:

File: 06-Customizing AzTS for your org/Extending AzTS/Readme.md [hunk 5]
```diff
@@ -2,19 +2,24 @@
...
 
-## Update existing control metadata
-   - [Update existing control metadata](UpdateControlMetadata.md)
+Once CMET is enabled, you can use below features:
+
+   - [Add new control based on custom Azure policy](/06-Customizing%20AzTS%20for%20your%20org/Extending%20AzTS/AddControlForPolicy.md)
+   - [Add new control based on MDC Assessment](/06-Customizing%20AzTS%20for%20your%20org/Extending%20AzTS/AddControlForAssessment.md) 
+   - [Update control metadata for existing control](../06-Customizing%20AzTS%20for%20your%20org/Extending%20AzTS/UpdateControlMetadata.md)
+
+   If you are looking for updating multiple controls at once, please follow steps mentioned [here](../06-Customizing%20AzTS%20for%20your%20org/Extending%20AzTS/FeaturesInCMET.md#bulk-edit)
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/Extending AzTS/Readme.md:20]: If you need to update multiple controls at once, please follow the steps outlined here:

File: 06-Customizing AzTS for your org/Extending AzTS/UpdateControlMetadata.md [hunk 6]
```diff
@@ -1,8 +1,8 @@
-## Update existig control metadata
-Control metadata like Display Name, Category, Tags, Description , Recommendations can be updated using Control Metadata Editor Tool(CMET). Please follow the below mentioned steps to update control metadata:
+## Update existing control metadata
+Control metadata like Display Name, Category, Tags, Description, Recommendations can be updated using Control Metadata Editor Tool(CMET). Please follow the below mentioned steps to update control metadata:
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/Extending AzTS/UpdateControlMetadata.md:2]: To update control metadata, including fields like Display Name, Category, Tags, Description, and Recommendations, you can utilize the Control Metadata Editor Tool (CMET). Follow the steps below to update control metadata:

File: 06-Customizing AzTS for your org/Extending AzTS/UpdateControlMetadata.md [hunk 7]
```diff
@@ -1,8 +1,8 @@
...
 
 **Step 0:** Following prerequisites are required:
-   1. CMET(Control metadata editor tool) is not enabled in AzTS setup by default. To enable this feature, please follow steps mentioned [here](Prerequisites.md#prerequisite-azts-configurations-to-enable-control-medatadata-editor-toolcmet).
+   1. CMET(Control metadata editor tool) is not enabled in AzTS setup by default. To enable this feature, please follow steps mentioned [here](Prerequisites.md#prerequisite-azts-configurations-to-enable-control-metadata-editor-toolcmet).
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/Extending AzTS/UpdateControlMetadata.md:5]: By default, this feature is not turned on. If you haven't enabled it in your AzTS setup yet, please follow the steps provided here to enable it.

File: 06-Customizing AzTS for your org/README.md [hunk 8]
```diff
@@ -3,13 +3,22 @@
 
 # Extending AzTS
 
-## Add new control for existing SVT
+## Customizing AzTS for your org
 
-Azure Tenant Security Solution (AzTS) provides capability to add new controls(for existing services supported by AzTS) to customize the AzTS for your organization as per your need.
-Currently you can extend AzTS controls set by either adding a new control based on custom Azure Policy or based on Microsoft Defender for Cloud assessment.
+### Control Metadata Editor tool
+CMET (Control Metadata Editor Tool) is a tool that can be used by privileged users (mostly by admins maintaining AzTS setup for organization)to customize AzTS for organization. It provides capabilities to update control metadata (Tags, Display Name, control settings(if applicable)), enable or disable control, etc.
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/README.md:9]: Pls. update as per the earlier recommendation. 
Same for other places where the common text/paragraph has been used.

File: 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+![Resources](../../Images/06_BulkEdit_2.png)
+
+
+### **Using Bulk Edit feature to add Custom tags to controls** 
+
+Once Bulk Edit Feature is enabled, follow below steps to add custom tag(s) for set of controls.
+
+1. Open AzTS UI in new browser window.
+2. Open CMET editor (for this user will require editor permission over CMET, if you don’t have permission, please follow steps mentioned [here](../Extending%20AzTS/Prerequisites.md#access-to-cmet-control-metadata-editor-tool)).
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md:30]: (To open the CMET editor, you need to have editor permissions. If you don't have the required permission, please follow the steps mentioned here)

File: 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+
+### **Validating custom tags added using Bulk Edit:**
+Once custom tag has been added for set of controls, follow below validation steps:
+
+1. Refresh CMET editor view.
+
+![Resources](../../Images/06_BulkEdit_Refresh.png)
+
+2. Search controls by newly added tag value (like ‘LearningBaselineTest’ in example), this will list all the controls which have this tag.
```
Comment (ZiaAhmad-PM) [inline 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md:85]: Shall we have this elaborated? 

To search controls by a newly added tag value, such as 'LearningBaselineTest' in this example, follow these steps:

Open the CMET editor.
Look for the search textbox within the editor.
Enter the tag value 'LearningBaselineTest' in the search field..
The CMET editor will display a list of controls that have the specified tag value, allowing you to view and manage them accordingly.
By following these steps, you can easily search for controls based on a specific tag value like 'LearningBaselineTest'.

File: Images/06_BulkEdit_MAUpdate_3.png
Comment (ZiaAhmad-PM) [inline Images/06_BulkEdit_MAUpdate_3.png:1]: Pls. underline using a scale/ruler.

File: Images/06_BulkEdit_MAUpdate_4.png
Comment (ZiaAhmad-PM) [inline Images/06_BulkEdit_MAUpdate_4.png:1]: Could we use paint instead to correctly box the intended actions. Like the orange ones, in other screenshots.

**Review by ZiaAhmad-PM — CHANGES_REQUESTED** [review]
Have reviewed and added the comments.

File: 06-Customizing AzTS for your org/Extending AzTS/AddControlForPolicy.md [hunk 11]
```diff
@@ -2,7 +2,7 @@
 Please follow the below mentioned steps to add new controls based on custom Azure Policy:
 
 **Step 0:** Following prerequisites are required to add new control.
```
[the remark this answers is not in the document]
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/AddControlForPolicy.md]: corrected

File: 06-Customizing AzTS for your org/Extending AzTS/AddControlForPolicy.md [hunk 12]
```diff
@@ -2,7 +2,7 @@
...
-   1. This feature is not enabled by default. If you have not enabled this feature in your AzTS setup yet, please follow steps mentioned [here](Prerequisites.md#prerequisite-azts-configurations-to-enable-control-medatadata-editor-toolcmet).
+   1. This feature is not enabled by default. If you have not enabled this feature in your AzTS setup yet, please follow steps mentioned [here](Prerequisites.md#prerequisite-azts-configurations-to-enable-control-metadata-editor-toolcmet).
```
[the remark this answers is not in the document]
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/AddControlForPolicy.md]: Updated

File: 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md (hunk 1 above)
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md]: Updated

File: 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md (hunk 2 above)
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md]: Updated

File: 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md [hunk 13]
```diff
@@ -1,7 +1,21 @@
-## Prerequisite AzTS configurations to enable Control Medatadata Editor tool(CMET)
+## Prerequisite AzTS configurations to enable Control Metadata Editor tool(CMET)
+
+
+Few application settings need to be configured for AzTS Function Apps/WebApps to enable Control Metadata Editor tool(CMET).These can be configured either from Azure portal or using the helper script. After these application settings are configured, please add users to Privileged editor roles using steps mentioned [here](#access-to-cmet-control-metadata-editor-tool).
```
[the remark this answers is not in the document]
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md]: Updated

File: 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md (hunk 3 above)
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/Prerequisites.md]: Updated

File: 06-Customizing AzTS for your org/Extending AzTS/Readme.md (hunk 5 above)
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/Readme.md]: Updated

File: 06-Customizing AzTS for your org/Extending AzTS/UpdateControlMetadata.md (hunk 6 above)
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/UpdateControlMetadata.md]: Updated

File: 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md (hunk 9 above)
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md]: Updated

File: 06-Customizing AzTS for your org/README.md (hunk 8 above)
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/README.md]: Updated

File: 06-Customizing AzTS for your org/Extending AzTS/Readme.md (hunk 4 above)
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/Readme.md]: Updated

File: 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
+## Features in CMET
+- ### Bulk Edit
+
+With Bulk Edit feature, custom tags can be added to more than 1 controls at a time.
```
[the remark this answers is not in the document]
Comment (Aboli-msft) [reply 06-Customizing AzTS for your org/Extending AzTS/FeaturesInCMET.md]: Updated

**Pull request merged**